Lib/test/test_clinic.py

```diff
@@ -2399,6 +2399,20 @@ class ClinicFunctionalTest(unittest.TestCase):
     locals().update((name, getattr(ac_tester, name))
                     for name in dir(ac_tester) if name.startswith('test_'))
 
+    def check_depr_star(self, *, fname, pnames, good_calls, bad_calls):
+        for call in good_calls:
+            call()
+        regex = (
+            f"Passing.*positional argument.*to {fname}.*is deprecated. "
+            f"Parameter.*{pnames} will become.*keyword-only parameter.*in "
+            "Python 3.14"
+        )
+        for call in bad_calls:
+            with self.assertWarnsRegex(DeprecationWarning, regex) as cm:
+                call()
+            self.assertTrue(cm.filename.endswith("test_clinic.py"), cm.filename)
```

Comment:
Also, since the tested function is called in a helper function, there are two frames with filename "test_clinic.py". To ensure that the warnings points at the right place you need to test either the line number (relatively to the line number of the helper function) or the line itself (the later will not work in a source-less distributive, but who distribute sourceless tests?).

Actually, there are three frames with the same filename, because you use lambdas.

Reply:
I added a test for the line number in cdf057442469080d48f51b01cb61e1706331a593